packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -0,0 +1,182 @@
+				newForm.errors)
+			)
+			.catch(({ errors }) => setError(formName, errors));
+	}
+
+	/**
+	 * Triggers submit callback if form is valid
+	 * @param event the submit event
+	 */
+	submit(event) {
+		event.preventDefault();
+		const { mergedSchema } = this.state;
+		const { formName, properties, customValidation } = this.props;
+		const errors = validateAll(mergedSchema, properties, customValidation);
+		this.props.setErrors(formName, errors);
+
+		const isValid = !Object.keys(errors).length;
+		if (isValid) {
+			this.props.onSubmit(event, properties);
+		}
+	}
+
+	render() {
+		const { autoComplete, errors, formName, id, properties, widgets } = this.props;
+		return (
+			<form
+				onSubmit={this.submit}
+				autoComplete={autoComplete}
+				noValidate
+			>
+				{
+					this.state.mergedSchema.map((nextSchema, index) => (
+						<Widget
+							id={id}
+							key={index}
+							formName={formName}
+							onChange={this.onChange}
+							onTrigger={this.onTrigger}
+							schema={nextSchema}
+							properties={properties}
+							errors={errors}
+							widgets={widgets}
+						/>
+					))
+				}
+				<button type="submit" className="btn btn-primary">Submit</button>
+			</form>
+		);
+	}
+}
+
+if (process.env.NODE_ENV !== 'production') {
+	UIForm.propTypes = {
+		/** Form auto complete */
+		autoComplete: PropTypes.bool,
+		/** Form definition: The form name that will be used to create ids */
+		formName: PropTypes.string,
+		/** The form id */
+		id: PropTypes.string,
+		/** Form definition: Json schema that specify the data model */
+		jsonSchema: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+		/**
+		 * Form definition: Form fields values.
+		 * Note that it should contains @definitionName for triggers.
+		 */
+		properties: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+		/** Form definition: UI schema that specify how to render the fields */
+		uiSchema: PropTypes.array, // eslint-disable-line react/forbid-prop-types
+		/** Form definition: The forms errors { [fieldKey]: errorMessage } */
+		errors: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+
+		/**
+		 * User callback: Custom validation function.
+		 * Prototype: function customValidation(schema, value, properties)
+		 * Return format : errorMessage String | falsy
+		 * This is triggered on fields that has their uiSchema > customValidation : true
+		 */
+		customValidation: PropTypes.func,
+		/** User callback: Form submit callback */
+		onSubmit: PropTypes.func.isRequired,
+		/**
+		 * User callback: Trigger > after callback.
+		 * Prototype: function onTrigger(type, schema, value, properties)
+		 * This is executed on changes on fields with uiSchema > triggers : ['after']
+		 */
+		onTrigger: PropTypes.func,
+		/** Custom widgets */
+		widgets: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+
+		/** State management impl: The change callback */
+		onChange: PropTypes.func.isRequired,
```

Comment:
Because the inputs are controlled inputs. We inject the value, we expect the container to manage the onChange and inject back the new values